guide/src/migration.md

```diff
@@ -198,6 +198,83 @@ impl<'a, 'py> IntoPyObject<'py> for &'a MyPyObjectWrapper {
 ```
 </details>
 
+### Free-threaded Python Support
+<details open>
+<summary><small>Click to expand</small></summary>
+
+PyO3 0.23 introduces preliminary support for the new free-threaded build of
+CPython 3.13. PyO3 features that implicitly assumed the existence of the GIL
+are not exposed in the free-threaded build, since they are no longer safe.
```

Comment:
In the end is `GILProtected` the only feature we're hiding? It looks like #4512 leaves `GILOnceCell` exposed.

Reply:
I'm considering #4512 and #4513 as possible ideas, would welcome your thoughts on those two options. It seems like migrating off `GILOnceCell` is desired in the long term, but I'd prefer not force users to do that until we can support module state...